### DISCUSSION TITLE
Debounce mechanism for accuracy

### DISCUSSION THREAD

**Discussion by hashimkhanzada** [vladmandic/human#427] [Q&A] [answered]
Hi,

I'm currently developing an internal face recognition system. It's works similarly to the face Id demo, but runs 24 / 7 and without most of the validation checks as it's meant to record them from a distance (about 2-3 meters).

The challenge I've got is the accuracy, especially when the person first enters the frame, when the lighting / angle isn't perfect. This usually returns an incorrect match from the database, while also returning a high score (close to 100%).

I was wondering if there's a built-in way for a 'debounce' mechanism. Essentially keeping track of the face to see if it's the same person for a short period (eg 500ms). The idea is that it would only confirm the recognition if it consistently identifies the person as the same individual throughout this duration.

If there's nothing built-in, could you recommend any workarounds for this? (eg should I just keep track of the faces with a timer and validate accordingly).

Also wondering if I should look into an alternative model to BlazeFace, as I'm assuming it's meant for more optimal conditions.

Thanks

**Comment by vladmandic** [comment, marked as answer]
there is no such built-in functionality, its expected that would be done in the app level.
regarding face match, blazeface is just a bounding box detector. actual face feature vector is performed using one of three options: facenet, insightface or mobilefacenet. facenet is default primarily because its fast and unobstrusive, not because its best.
see https://github.com/vladmandic/human/blob/main/demo/faceid/index.ts for example.

**Comment by hashimkhanzada** [comment]
Thanks for your response. 

Also with regards to the config, I'm using angular with typescript.
I have the config set to type 'Config', and it doesn't seem to recognize 'insightface' as a field. Is this outdated or am I supposed to set the modelPath in the 'description' instead, to override the default (as below).
It does seem to load fine according to the logs, but once it gets to the warmup, it throws the error attached.

I downloaded the 'insightface-mobilenet-swish.json' and 'insightface-mobilenet-swish.bin' from the github url in the config below, and put them in the face-models folder.

>   humanConfig: Partial<Config> = {
    // user configuration for human, used to fine-tune behavior
    cacheSensitivity: 1,
    modelBasePath: '../../../../assets/face-models/',
    filter: { enabled: true, equalization: true }, // lets run with histogram equilizer
    debug: true,
    backend: 'webgl',
    face: {
      enabled: true,
      detector: { rotation: true, return: true, mask: false, maxDetected: 5 }, // return tensor is used to get detected face image
      mesh: { enabled: true },
      description: {
        enabled: true,
        modelPath:
          '../../../../assets/face-models/insightface-mobilenet-swish.json',
      }, // default model for face descriptor extraction is faceres
      // mobilefacenet: { enabled: true, modelPath: 'https://vladmandic.github.io/human-models/models/mobilefacenet.json' }, // alternative model
      // insightface: { enabled: true, modelPath: 'https://vladmandic.github.io/insightface/models/insightface-mobilenet-swish.json' }, // alternative model
      iris: { enabled: false }, // needed to determine gaze direction
      emotion: { enabled: false }, // not needed
      antispoof: { enabled: false }, // enable optional antispoof module
      liveness: { enabled: false }, // enable optional liveness module
    },
    body: { enabled: false },
    hand: { enabled: false },
    object: { enabled: false },
    gesture: { enabled: false }, // parses face and iris gestures
  };

![image](https://github.com/vladmandic/human/assets/74533711/eb13e7e7-7288-4220-9b49-291ef1280116)

**Comment by vladmandic** [comment]
i'll double-check
